stores/sql/main.go

```diff
@@ -782,6 +788,31 @@ func Hosts(ctx context.Context, tx sql.Tx, autopilot, filterMode, usabilityMode,
 	if hasBlocklist {
 		blockedExprs = append(blockedExprs, "EXISTS (SELECT 1 FROM host_blocklist_entry_hosts hbeh WHERE hbeh.db_host_id = h.id)")
 	}
+
+	var orderByExpr string
+	if opts.SortBy != "" {
+		switch opts.SortDir {
+		case "", api.SortDirAsc, api.SortDirDesc:
+		default:
+			return nil, fmt.Errorf("invalid sort order: %v", opts.SortDir)
+		}
+		if !api.IsValidHostSortBy(opts.SortBy) {
+			return nil, fmt.Errorf("%w: %s", api.ErrInvalidHostSortBy, opts.SortBy)
+		}
+
+		var fieldExpr string
+		if strings.HasPrefix(opts.SortBy, "settings.") {
+			field := strings.TrimPrefix(opts.SortBy, "settings.")
+			fieldExpr = fmt.Sprintf("h.settings ->> '$.%s'", field)
+		} else if strings.HasPrefix(opts.SortBy, "price_table.") {
+			field := strings.TrimPrefix(opts.SortBy, "price_table.")
+			fieldExpr = fmt.Sprintf("h.price_table ->> '$.%s'", field)
+		} else {
+			return nil, fmt.Errorf("invalid sortBy parameter: %v", opts.SortBy)
+		}
+		orderByExpr = fmt.Sprintf("ORDER BY %s %s", fieldExpr, opts.SortDir)
```

Comment:
should we default the `SortDir` to `asc` if it's empty?

Reply:
Strictly speaking it does default to ascending but we can be explicit about it to not leave it up to the db engine